ci/cudf_pandas_scripts/pandas-tests/run.sh

```diff
@@ -1,12 +1,14 @@
 #!/usr/bin/env bash
-# SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: Copyright (c) 2023-2024, NVIDIA CORPORATION & AFFILIATES.
 # All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 
+set -euo pipefail
+
 PANDAS_TESTS_BRANCH=${1}
 
 rapids-logger "Running Pandas tests using $PANDAS_TESTS_BRANCH branch"
-rapids-logger "PR number: $RAPIDS_REF_NAME"
+rapids-logger "PR number: ${RAPIDS_REF_NAME:-"unknown"}"
```

Comment:
Making this conditional makes it easier to repro CI locally. Otherwise you get an unbound variable error.